httpserver/server.go

```diff
@@ -192,3 +189,34 @@ type ServerAddrNotFoundError struct{}
 func (err ServerAddrNotFoundError) Error() string {
 	return "The http server address configuration is not set. Skipping server creation"
 }
+
+func serverGracefulShutdown(isTLSEnabled bool, svr *http.Server, lsnr net.Listener, certFile string, keyFile string) error {
```

Comment:
suggestion to rename to indicate this method also handles server startup. General suggestion to add method and subroutine description as well :)